src/System.Runtime.Extensions/src/System/IO/Path.cs

```diff
@@ -214,6 +217,20 @@ public static string Combine(string path1, string path2, string path3)
             return CombineNoChecks(path1, path2, path3);
         }
 
+        public static string Combine(string path1, string path2, string path3, string path4)
+        {
+            if (path1 == null || path2 == null || path3 == null || path4 == null)
+                throw new ArgumentNullException((path1 == null) ? "path1" : (path2 == null) ? "path2" : (path3 == null) ? "path3" : "path4");
+            Contract.EndContractBlock();
```

Comment:
>Contract.EndContractBlock(); [](start = 12, length = 28)

are we still using Contract? or just to be consistent with the other methods here?